packages/pwa/app/components/locale-selector-text/index.jsx

```diff
@@ -0,0 +1,109 @@
+/*
```

Comment:
First thing, right off the bat, I find it a little weird to have a component names "SomethingSelectorText" being used outside of the context of an actual select. But don't worry I didn't come without a suggestion, this is what I would do:

Rename the component to `<LocaleText>` and the the props accepted are simply `shortCode` and `as`. This way if you have the flexibility to use the component in all kind of situations, here are our examples below:

_Mobile Menu_
```
<Accordion>
    <AccordionItem>
        <LocaleText shortCode="en-GB" />
    </AccordionItem>
</Accordion>
```

_Desktop Footer_
```
<select>
    {
        SUPPORTED_LOCALES.map(({shortCode}) => (
            <LocaleText shortCode={shortCode} as="options" value={shortCode} />
        ))
    }
</select>
```

You'll notice that you'll have to pass the `rest` props down to the rendered component as you are now. But thats not a big deal.

So what might the `LocaleText` component look like? (This is a simple definition of it)
```
import {chakra} from '@chakra/system'

const LocaleText = ({shortCode, as, ...otherProps}) => {
    const Wrapper = as ? chakra(as) : Text

    // Your logic for finding the text and handling     
    // NOTE: I feel like you can DRY up your code with respect to displaying the message or the fallback
    // I haven't played around with it tho. Its typically not normal to use the intl.formateMessage and the 
    // component at the same time.

    return <Wrapper>{calculatedMessage}</Wrapper>
}
```

Reply:
It's cool that there's the factory function `chakra`. For more details: https://chakra-ui.com/docs/features/chakra-factory